massmailer/forms.py

```diff
@@ -110,14 +111,33 @@ def foolproof_field(cls, count):
     def foolproof_enabled(self):
         return not settings.DEBUG
 
+    def clean(self):
+        query = self.cleaned_data['query']
+        result, qs = massmailer.models.Query.execute(query.query)
+        template = self.cleaned_data['template']
+        if (
+            len(qs) > 0
```

Comment:
Fail validation if qs is empty. An empty batch doesn't make sense and in this case you need an example email to validate further.